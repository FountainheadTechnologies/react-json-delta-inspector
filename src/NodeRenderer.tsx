import * as React from 'react';
import { ObjectLabel } from 'react-inspector';

import { ObjectRootLabel } from './ObjectRootLabel';
import { styles } from './styles';

export type Props = {
  depth: number;
  name?: string;
  data: {};
  diff?: keyof typeof styles;
};

/**
 * A re-implementation of the `react-inspector` nodeRenderer, except that it
 * uses our own version of `ObjectRootLabel`, and wraps `ObjectLabel` in a span
 * that is styled according to the diff operation that occurred to this value,
 * if any.
 */
export const NodeRenderer: React.SFC<Props> = ({ depth, name, data, diff }) =>
  depth === 0 ?
    (
      <ObjectRootLabel name={name} data={data} />
    ) : (
      <span style={diff ? styles[diff] : {}}>
        <ObjectLabel name={name} data={data} isNonenumerable={false} />
      </span>
    );
