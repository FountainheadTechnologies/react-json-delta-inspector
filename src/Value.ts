/**
 * These classes simply act as 'containers' for values that have been added,
 * deleted or changed in an object.
 */
export class DeletedValue {
  constructor(public prevVal: any) { }
}

// tslint:disable-next-line:max-classes-per-file
export class ChangedValue {
  constructor(public prevVal: any, public newVal: any) { }
}

// tslint:disable-next-line:max-classes-per-file
export class AddedValue {
  constructor(public val: any) { }
}
