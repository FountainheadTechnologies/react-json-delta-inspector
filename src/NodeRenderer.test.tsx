import * as React from 'react';

import { configure, shallow } from 'enzyme';
import * as EnzymeAdapter from 'enzyme-adapter-react-16';

import { NodeRenderer } from './NodeRenderer';
import { AddedValue, ChangedValue, DeletedValue } from './Value';

configure({ adapter: new EnzymeAdapter() });

const diffObject = {
  deletedString: new DeletedValue('string'),
  addedNumber: new AddedValue(10),
  array: [
    'zero',
    'one',
    new ChangedValue('too', 'two')
  ]
};

describe('NodeRenderer', () => {
  it('renders the root object', () => {
    const wrapper = shallow(
      <NodeRenderer
        depth={0}
        data={diffObject}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders each item in the object', () => {
    (Object.keys(diffObject) as Array<keyof typeof diffObject>).forEach(key => {
      const wrapper = shallow(
        <NodeRenderer
          depth={1}
          data={diffObject[key]}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
