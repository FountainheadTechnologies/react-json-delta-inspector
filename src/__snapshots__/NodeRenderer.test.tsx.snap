// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeRenderer renders each item in the object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
      DeletedValue {
        "prevVal": "string",
      }
    }
    depth={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ObjectLabel
        data={
          DeletedValue {
            "prevVal": "string",
          }
        }
        isNonenumerable={false}
      />,
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": DeletedValue {
          "prevVal": "string",
        },
        "isNonenumerable": false,
        "name": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ObjectLabel
          data={
            DeletedValue {
              "prevVal": "string",
            }
          }
          isNonenumerable={false}
        />,
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": DeletedValue {
            "prevVal": "string",
          },
          "isNonenumerable": false,
          "name": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`NodeRenderer renders each item in the object 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
      AddedValue {
        "val": 10,
      }
    }
    depth={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ObjectLabel
        data={
          AddedValue {
            "val": 10,
          }
        }
        isNonenumerable={false}
      />,
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": AddedValue {
          "val": 10,
        },
        "isNonenumerable": false,
        "name": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ObjectLabel
          data={
            AddedValue {
              "val": 10,
            }
          }
          isNonenumerable={false}
        />,
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": AddedValue {
            "val": 10,
          },
          "isNonenumerable": false,
          "name": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`NodeRenderer renders each item in the object 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
      Array [
        "zero",
        "one",
        ChangedValue {
          "newVal": "two",
          "prevVal": "too",
        },
      ]
    }
    depth={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ObjectLabel
        data={
          Array [
            "zero",
            "one",
            ChangedValue {
              "newVal": "two",
              "prevVal": "too",
            },
          ]
        }
        isNonenumerable={false}
      />,
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Array [
          "zero",
          "one",
          ChangedValue {
            "newVal": "two",
            "prevVal": "too",
          },
        ],
        "isNonenumerable": false,
        "name": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ObjectLabel
          data={
            Array [
              "zero",
              "one",
              ChangedValue {
                "newVal": "two",
                "prevVal": "too",
              },
            ]
          }
          isNonenumerable={false}
        />,
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [
            "zero",
            "one",
            ChangedValue {
              "newVal": "two",
              "prevVal": "too",
            },
          ],
          "isNonenumerable": false,
          "name": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`NodeRenderer renders the root object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
      Object {
        "addedNumber": AddedValue {
          "val": 10,
        },
        "array": Array [
          "zero",
          "one",
          ChangedValue {
            "newVal": "two",
            "prevVal": "too",
          },
        ],
        "deletedString": DeletedValue {
          "prevVal": "string",
        },
      }
    }
    depth={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Object {
        "addedNumber": AddedValue {
          "val": 10,
        },
        "array": Array [
          "zero",
          "one",
          ChangedValue {
            "newVal": "two",
            "prevVal": "too",
          },
        ],
        "deletedString": DeletedValue {
          "prevVal": "string",
        },
      },
      "name": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "addedNumber": AddedValue {
            "val": 10,
          },
          "array": Array [
            "zero",
            "one",
            ChangedValue {
              "newVal": "two",
              "prevVal": "too",
            },
          ],
          "deletedString": DeletedValue {
            "prevVal": "string",
          },
        },
        "name": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
